nifi-frontend/src/main/frontend/apps/nifi-registry/src/app/pages/expolorer/feature/ui/droplet-table-filter/droplet-table-filter.component.ts

```diff
@@ -0,0 +1,164 @@
+import { MatSelectModule } from '@angular/material/select';
+import { MatInputModule } from '@angular/material/input';
+import { Bucket } from 'apps/nifi-registry/src/app/state/buckets';
+import { debounceTime } from 'rxjs';
+import { takeUntilDestroyed } from '@angular/core/rxjs-interop';
+import { MatButtonModule } from '@angular/material/button';
+
+export interface DropletTableFilterColumn {
+    key: string;
+    label: string;
+}
+
+export interface DropletTableFilterArgs {
+    filterTerm: string;
+    filterColumn: string;
+    filterBucket?: string;
+}
+
+export interface DropletTableFilterContext extends DropletTableFilterArgs {
+    changedField: string;
+}
+
+@Component({
+    selector: 'droplet-table-filter',
+    standalone: true,
+    imports: [CommonModule, ReactiveFormsModule, MatFormFieldModule, MatSelectModule, MatInputModule, MatButtonModule],
+    templateUrl: './droplet-table-filter.component.html',
+    styleUrl: './droplet-table-filter.component.scss'
+})
+export class DropletTableFilterComponent implements AfterViewInit {
+    filterForm: FormGroup;
+    private _initialFilterColumn = 'name';
+    private _filterableColumns: DropletTableFilterColumn[] = [];
+    private _buckets: Bucket[] = [];
+    private _filteredCount = 0;
+    private _totalCount = 0;
+    private destroyRef: DestroyRef = inject(DestroyRef);
+
+    set filterableColumns(filterableColumns: DropletTableFilterColumn[]) {
+        this._filterableColumns = filterableColumns;
+    }
+    get filterableColumns(): DropletTableFilterColumn[] {
+        return this._filterableColumns;
+    }
+
+    @Input() set buckets(filterableColumns: Bucket[]) {
+        this._buckets = filterableColumns;
+    }
+    get buckets(): Bucket[] {
+        return this._buckets;
+    }
+
+    @Input() set filterTerm(term: string) {
+        this.filterForm.controls['filterTerm']?.setValue(term);
+    }
+
+    @Input() set filterColumn(column: string) {
+        this._initialFilterColumn = column;
+        if (this.filterableColumns?.length > 0) {
+            if (this.filterableColumns.findIndex((col) => col.key === column) >= 0) {
+                this.filterForm.get('filterColumn')?.setValue(column);
+            } else {
+                this.filterForm.get('filterColumn')?.setValue(this.filterableColumns[0].key);
+            }
+        } else {
+            this.filterForm.get('filterColumn')?.setValue(this._initialFilterColumn);
+        }
+    }
+
+    @Input() set filterBucket(term: string) {
+        this.filterForm.controls['filterBucket']?.setValue(term);
+    }
+
+    @Input() set filteredCount(filteredCount: number) {
+        this._filteredCount = filteredCount;
+    }
+
+    get filteredCount(): number {
+        return this._filteredCount;
+    }
+
+    @Input() set totalCount(totalCount: number) {
+        this._totalCount = totalCount;
+    }
+
+    get totalCount(): number {
+        return this._totalCount;
+    }
```

Comment:
what is the purpose to use getter/setter here instead of `@Input() totalCount: number`